### PR TITLE
fix flaky TestFastDisconnect identify test

### LINKED ISSUE #1157: flaky FastDisconnect test
```
  === RUN   TestFastDisconnect
      id_glass_test.go:77: context deadline exceeded
  --- FAIL: TestFastDisconnect (5.00s)

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #1157.

### CODE REVIEW

File: p2p/protocol/identify/id_glass_test.go [hunk 1]
```diff
@@ -36,10 +37,11 @@ func TestFastDisconnect(t *testing.T) {
 
 		// Kill the connection, and make sure we're completely disconnected.
 		s.Conn().Close()
```
Comment (marten-seemann) [inline p2p/protocol/identify/id_glass_test.go:39]: The problem here was that _sometimes_, we'd end up with two connections (for example, one TCP and one QUIC connection). Of course, calling `s.Conn().Close()` here would only close one of them, which is why the loop on the next line would never terminate.

Comment (marten-seemann) [reply p2p/protocol/identify/id_glass_test.go:39]: Alternative fix would be to loop over all connections and close them.

Comment (vyzo) [reply p2p/protocol/identify/id_glass_test.go:39]: actually, let's do that -- it's the right thing to do.

Comment (marten-seemann) [reply p2p/protocol/identify/id_glass_test.go]: ok, done

**Review by vyzo — APPROVED** [review]

**Pull request merged**